### PR TITLE
Guard UsaSpa and add default handler

### PR DESCRIPTION
**Pull request by cailyncodes**

### CODE REVIEW

File: src/Hedwig/Startup.cs [hunk 1]
```diff
@@ -79,52 +84,64 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
...
 
-			app.UseSpa(spa =>
-			{
-				spa.Options.SourcePath = "ClientApp";
-
-				// If we are development, that means we aren't serving SPA files
-				// as static, compiled resources. So we need to forward requests a
-				// development server.
-				if (env.IsDevelopment())
+			// UseSpa internally calls UseStaticFiles which only supports GET or OPTIONS
```
Comment (jlhogan) [inline src/Hedwig/Startup.cs:98]: tsk tsk useSpa maintainers!

File: src/Hedwig/Startup.cs [hunk 2]
```diff
@@ -79,52 +84,64 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
...
-				}
-			});
 
+					// If we aren't in development, UseSpa will internally call UseStaticFiles
+					// on the directory specified by AddSpaStaticFiles
+				})
+			);
+
+			// Catch any remaining requests and return 404 not found
+			app.Run(
```
Comment (jlhogan) [inline src/Hedwig/Startup.cs:139]: does this `Run()` only run after the above `UseWhen()` ? im confused about the request flow here. what happens to the PUT/POST/DELETES?

Comment (cailyncodes) [reply src/Hedwig/Startup.cs:139]: GET/HEAD will fall through the UseSpa above.

PUT/POST/DELETE/OPTIONS will not match the UseWhen so will continue to the app.Run

Comment (jlhogan) [reply src/Hedwig/Startup.cs:139]: got it! forgot abt the endpoint delegates set up by the controllers.

File: src/Hedwig/Startup.cs [hunk 3]
```diff
@@ -79,52 +84,64 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
...
+							// script and provided the referenced executable exists on the path
+							// the script will run.
+							spa.UseReactDevelopmentServer(npmScript: "start");
+						}
 					}
-				}
-			});
 
+					// If we aren't in development, UseSpa will internally call UseStaticFiles
+					// on the directory specified by AddSpaStaticFiles
```
Comment (jlhogan) [inline src/Hedwig/Startup.cs:134]: does setting `spa.Options.SourcePath` implicitly do the same thing as `AddSpaStaticFiles`?

Comment (cailyncodes) [reply src/Hedwig/Startup.cs:134]: Nope -- that tells the UseReactServer where the code lives to serve

Comment (cailyncodes) [reply src/Hedwig/Startup.cs:134]: UseSpa does the same logic as AddSpaStaticFiles under the scenes

Comment (cailyncodes) [reply src/Hedwig/Startup.cs:134]: AddSpaStaticFiles is necessary because it updates DI container for use in UseSpa

Comment (jlhogan) [reply src/Hedwig/Startup.cs:134]: got confused b/c `UseSpaStaticFiles` went away, but `AddSpaStaticFiles` didn't show up. see now it's called in service extension method.

**Pull request merged**